src/script/interpreter.cpp

```diff
@@ -1781,6 +2062,222 @@ bool GenericTransactionSignatureChecker<T>::CheckSequence(const CScriptNum& nSeq
+    ScriptExecutionData& execdata,
+    const size_t trigger_out_idx,
+    const int maybe_revault_out_idx,
+    const CAmount revault_amount,
+    CScript flu_script,
+    unsigned int flags,
+    ScriptError* serror) const
+{
+    assert(execdata.m_curr_witversion);
+
+    // Implement the sigops/witnesssize ratio test.
+    // Passing with an upgradable public key version is also counted.
+    assert(execdata.m_validation_weight_left_init);
+    execdata.m_validation_weight_left -= VALIDATION_WEIGHT_PER_TAPTWEAKCHECK;
+    if (execdata.m_validation_weight_left < 0) {
+        return SCRIPT_ERR_TAPSCRIPT_VALIDATION_WEIGHT;
+    }
+
+    if (!this->txdata) {
+        HandleMissingData(m_mdb);
+        return SCRIPT_ERR_UNKNOWN_ERROR;
+    }
+    const auto& txd{*this->txdata};
+    const auto& vout{this->txTo->vout};
+
+    assert(txd.m_spent_outputs_ready);
+    const CScript& vault_spk = txd.m_spent_outputs[this->nIn].scriptPubKey;
+
+    if (trigger_out_idx >= vout.size()) {
+        return SCRIPT_ERR_UNVAULT_MISMATCH;
+    }
+
+    // Ensure that the output creating the unvault trigger matches the
+    // expected scriptPubKey.
+    const auto& value_out = vout[trigger_out_idx];
+
+    int trigger_witversion;
+    valtype trigger_witprogram;
+    const bool is_unvault_output_wit =
+        value_out.scriptPubKey.IsWitnessProgram(trigger_witversion, trigger_witprogram);
+
+    if (!is_unvault_output_wit || trigger_witversion < 1 ||
+            trigger_witversion != *execdata.m_curr_witversion) {
+        return SCRIPT_ERR_UNVAULT_INCOMPAT_OUTPUT_TYPE;
+    }
+    else if (trigger_witversion == 1) {
+        assert(execdata.m_internal_key);
+        assert(execdata.m_taproot_control.size() > 0);
+
+        const valtype& control = execdata.m_taproot_control;
+        const valtype val_sPK{
+            value_out.scriptPubKey.begin(), value_out.scriptPubKey.end()};
+
+        if (val_sPK.size() != 34) {
+            return SCRIPT_ERR_UNVAULT_MISMATCH;
+        }
+        const XOnlyPubKey val_pk{Span{val_sPK}.subspan(2)};
+
+        const auto tapleaf_hash = ComputeTapleafHash(
+            control[0] & TAPROOT_LEAF_MASK, flu_script);
+        const uint256 merkle_root = ComputeTaprootMerkleRoot(control, tapleaf_hash);
+        const auto expected_pk = (*execdata.m_internal_key).CreateTapTweak(&merkle_root);
+
+        // Ensure the currently executing OP_VAULT tapleaf is substituted for the
+        // expected replacement.
+        if (!(expected_pk && val_pk == expected_pk->first)) {
+            return SCRIPT_ERR_UNVAULT_MISMATCH;
+        }
+    }
+    // Append the handling of future witness versions with `else if`s here.
+    else {
+        if (flags & SCRIPT_VERIFY_DISCOURAGE_UPGRADABLE_WITNESS_PROGRAM) {
+            return SCRIPT_ERR_DISCOURAGE_UPGRADABLE_WITNESS_PROGRAM;
+        }
+        set_success(serror);
+        return std::nullopt;
+    }
```

Comment:
This is unreachable? Since OP_VAULT is only defined for v1 and we don't know what v2 could even look like this might as well not introduce the `m_curr_witversion` variable and explicitly check for witness version 1 above?

(was just checking the code corresponding to https://github.com/bitcoin/bips/pull/1421#discussion_r1345761010)

Reply:
Fixed, thanks.